### PR TITLE
[Doc] Add KoP authentication

### PR DESCRIPTION
**Pull request by Anonymitaet**
Doc architecture

image

### CODE REVIEW

**Comment by Anonymitaet** [comment]
@BewareMyPower @Jennifer88huang @Huanli-Meng could you please help review? Thanks

File: docs/security.md [hunk 1]
```diff
@@ -1,60 +1,259 @@
...
+
+    To get an access token from an OAuth 2.0 server, you need to use the KoP built-in callback handler instead of the Kafka login callback handler.
+
+    ```bash
+    mvn clean install -pl oauth-client -DskipTests
+    ```
+
+    (2) Add the following dependencies to the `pom.xml` file.
+
+    For stable releases, the `pulsar.version` is the same to the `kop.version`.
```
Comment (Jennifer88huang) [inline docs/security.md:165]: the same as?

**Review by Jennifer88huang — APPROVED** [review]
approve from language part

File: docs/security.md (hunk 1 above)
Comment (Anonymitaet) [reply docs/security.md:165]: image

Think `same to` is OK

File: docs/security.md [hunk 2]
```diff
@@ -1,60 +1,259 @@
...
+            </exclusion>
+        </exclusions>
+        </dependency>
+
+        <!-- KoP's login callback handler has a pulsar-client dependency -->
+        <dependency>
+        <groupId>org.apache.pulsar</groupId>
+        <artifactId>pulsar-client</artifactId>
+        <version>${pulsar.version}</version>
+        </dependency>
```
Comment (BewareMyPower) [inline docs/security.md:185]:
```suggestion
    <dependency>
      <groupId>org.apache.pulsar</groupId>
      <artifactId>pulsar-client</artifactId>
      <version>2.8.0-SNAPSHOT</version>
    </dependency>
    <dependency>
      <groupId>io.streamnative.pulsar.handlers</groupId>
      <artifactId>oauth-client</artifactId>
      <version>${kop.version}</version>
      <exclusions>
        <exclusion>
          <groupId>org.apache.logging.log4j</groupId>
          <artifactId>log4j-slf4j-impl</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

    <!-- KoP's login callback handler has a pulsar-client dependency -->
    <dependency>
      <groupId>org.apache.pulsar</groupId>
      <artifactId>pulsar-client</artifactId>
      <version>${pulsar.version}</version>
    </dependency>
```

Add code intent.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: docs/security.md (hunk 2 above)
Comment (BewareMyPower) [inline docs/security.md:185]:
```suggestion
    <dependency>
      <groupId>io.streamnative.pulsar.handlers</groupId>
      <artifactId>oauth-client</artifactId>
      <version>${kop.version}</version>
      <exclusions>
        <exclusion>
          <groupId>org.apache.logging.log4j</groupId>
          <artifactId>log4j-slf4j-impl</artifactId>
        </exclusion>
      </exclusions>
    </dependency>

    <!-- KoP's login callback handler has a pulsar-client dependency -->
    <dependency>
      <groupId>org.apache.pulsar</groupId>
      <artifactId>pulsar-client</artifactId>
      <version>${pulsar.version}</version>
    </dependency>
```

Add code intent. (The previous request change is wrong).

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: docs/security.md [hunk 3]
```diff
@@ -1,60 +1,259 @@
...
+
+    To forward your credentials, `SASL-PLAIN` is used on the Kafka client side. To enable `SASL-PLAIN`, you need to set the following properties through Kafka JAAS.
+
+    Property | Description | Example value
+    |---|---|---
+    `username` | `username` of Kafka JAAS is the `tenant/namespace`, where Kafka’s topics are stored in Pulsar.|`public/default`
+    `password`|`password` must be your token authentication parameters from Pulsar.<br><br>The token can be created by Pulsar token tools. The role is the `subject` for the token. It is embedded in the created token and the broker can get `role` by parsing this token.<br><br> **Note**: make sure the role of `password` has the permission to produce or consume the namespace of `username`. For more information, see [Authorization](http://pulsar.apache.org/docs/en/security-jwt/#authorization).|`token:xxx`
+
+    ```properties
+    security.protocol=SASL_PLAINTEXT # or SASL_SSL if SSL connection is used
```
Comment (BewareMyPower) [inline docs/security.md:87]:
```suggestion
    security.protocol=SASL_PLAINTEXT  # or security.protocol=SASL_SSL if SSL connection is used
```

Keep it consistent with the change in `OAUTHBEARER` section.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

**Review by BewareMyPower — APPROVED** [review]

**Pull request merged**